tsdb/head_wal.go

```diff
@@ -499,7 +499,17 @@ func (h *Head) resetSeriesWithMMappedChunks(mSeries *memSeries, mmc, oooMmc []*m
 	h.metrics.chunksRemoved.Add(float64(len(mSeries.mmappedChunks)))
 	h.metrics.chunks.Add(float64(len(mmc) + len(oooMmc) - len(mSeries.mmappedChunks)))
 	mSeries.mmappedChunks = mmc
-	mSeries.oooMmappedChunks = oooMmc
+	mSeries.ooo = nil
+	if len(oooMmc) == 0 {
+		mSeries.ooo = nil
+	} else {
+		if mSeries.ooo == nil {
+			mSeries.ooo = &memSeriesOOOFields{}
+		}
+		mSeries.ooo.oooHeadChunk = nil
+		mSeries.ooo.firstOOOChunkID = 0
+		mSeries.ooo.oooMmappedChunks = oooMmc
```

Comment:
This could be `*mSeries.ooo = memSeriesOOOFields{ oooMmappedChunks: oooMmc }`, which is more resilient against fields being added.

Reply:
Done!